cli/commands/preview/list.ts

```diff
@@ -0,0 +1,65 @@
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { getSiteIdFromFolder, readAppdata, Snapshot } from 'cli/lib/appdata';
+import { getAuthToken } from 'cli/lib/auth';
+import { getSnapshotCliTable } from 'cli/lib/output';
+import { validateSiteFolder } from 'cli/lib/validation';
+import { Logger, LoggerError } from 'cli/logger';
+import { RegisterCommand, OutputFormat } from 'cli/types';
+
+export enum LoggerAction {
+	VALIDATE = 'validate',
+	LOAD = 'load',
+}
+
+async function getSnapshots( userId: number, siteFolder: string ): Promise< Snapshot[] > {
+	const siteId = await getSiteIdFromFolder( siteFolder );
+	const userData = await readAppdata();
+	const snapshots = userData.snapshots ?? [];
+
+	return snapshots
+		.filter( ( snapshot ) => snapshot.userId === userId )
+		.filter( ( snapshot ) => snapshot.localSiteId === siteId );
+}
+
+async function runCommand( siteFolder: string, outputFormat?: OutputFormat ): Promise< void > {
+	const logger = new Logger< LoggerAction >( outputFormat );
+
+	try {
+		logger.reportStart( LoggerAction.VALIDATE, 'Validating...' );
+		validateSiteFolder( siteFolder );
+		const token = await getAuthToken();
+		logger.reportSuccess( 'Validation successful' );
+
+		logger.reportStart( LoggerAction.LOAD, __( 'Loading snapshots...' ) );
```

Comment:
Nit: If the above is false, then do we need a translated string for the logger messages?